### DISCUSSION TITLE
Speex vs rnnoise? Difference between these two noise-reduction plugins?

### DISCUSSION THREAD

**Discussion by 1player** [wwmm/easyeffects#1992] [General]
I've just updated to 7.0, and saw there is a new speex noise reduction plugin. Can we have more details on the difference between this one and the old rnnoise effect?

I am aware rnnoise is pretty much unmaintained, but I'd like to know whether Speex is a drop-in replacement, and what are the pros and cons of using one or the other. rnnoise has been decent enough and a mainstay of my input pipeline, but I admit it still is nowhere as good as NVIDIA Broadcast on Windows...

**Comment by wwmm** [comment]
>I'd like to know whether Speex is a drop-in replacement

I do not see it as a drop-in replacement. Rnnoise is based on a neural network. When the noise matches the one rnnoise's model was trained for it works really well. But when the noise does not match rnnoise often removes the voice and the noise. I do not know what kind of algorith speex is using but it seems to be more "general purpose". Usually it does not fully remove the noise like rnnoise is frequently capable of doing. But I have seen it suppression noise in situations where rnnoise does not work well.

On top of that speex offers a voice detection algorithm that has been working very well at least for me. So depending on the situation it makes sense to use both plugins at the same time.
